### PR TITLE
Adds Map preprocessing configuration

### PR DESCRIPTION
**Pull request by mustafaiman**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/MapConfig.java [hunk 1]
```diff
@@ -703,6 +711,21 @@ public MapConfig setMapAttributeConfigs(List<MapAttributeConfig> mapAttributeCon
         return this;
     }
 
+    public PreprocessingPolicy getPreprocessingPolicy() {
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/config/MapConfig.java]: Can you add javadoc?

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 2]
```diff
@@ -845,6 +845,11 @@
         rely on it. If true, it increases the speed of query processes in the map. It only works when
         <in-memory-format> is set to BINARY and performs a pre-caching on the entries queried. Its default
         value is false.
+        * <preprocessing-policy>
+        Pre-processing policy for this map.
+        Valid values are:
+        CREATION_TIME (default): Objects of supported types are pre-processed when they are created and updated.
```
Comment (mmedenjak) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: Default is `OFF`. Also, maybe add some link or additional comment as to what kind of pre-processing is being done.
Same in `hazelcast-default.xml`, `hazelcast-config-3.12.xsd` and `hazelcast-fullconfig.xml`.

Comment (mustafaiman) [reply hazelcast/src/main/resources/hazelcast-full-example.xml]: good catch. Thanks

**Comment by mustafaiman** [comment]
Thanks for reviews @mmedenjak and @blazember . @mmedenjak  can you take another look.

**Pull request merged**